packages/gatsby/src/schema/infer-graphql-input-fields.js

```diff
@@ -205,6 +205,7 @@ export function inferInputObjectStructureFromNodes({
       prefix: `${prefix}${_.upperFirst(key)}`,
     })
 
+    if (typeof key !== 'string') return
```

Comment:
:-)

Gatsby auto-creates a GraphQL schema from your data. Part of which is generating the field names / keys for the various "objects" in the schema. Keys have to be strings that start with [a-zA-Z].

Source/transformer plugins have to ensure this is the case.